src/syntax/syntax.rkt

```diff
@@ -71,53 +80,81 @@
 
 (define (operator-info operator field) (table-ref operators operator field))
 
+(define (operator-remove! operator)
+  (table-remove! operators operator)
+  (*loaded-ops* (remove operator (*loaded-ops*))))
+
+(define (prune-operators!)
+  (unless (flag-set? 'precision 'fallback)
+    (for ([op (if (flag-set? 'precision 'double) (*unknown-d-ops*) (*unknown-f-ops*))])
+      (operator-remove! op)))
+
+  (unless (flag-set? 'fn 'cbrt) (operator-remove! 'cbrt)))
+
 (define-syntax-rule (define-operator (operator atypes ...) rtype [key value] ...)
-  (let ([type (hash (length '(atypes ...)) (list (list '(atypes ...) 'rtype)))]
-        [args (list (length '(atypes ...)))])
+  (let ([type (hash (length '(atypes ...)) (list '(atypes ...) 'rtype))]
+        [args (list (length '(atypes ...)))]
+        [nonffi (if (findf (λ (x) (equal? (car x) 'nonffi)) '([key value] ...))
```

Comment:
Is the idea here to make the `nonffi` field an optional field? Instead of doing that, I think it'd be better to name the lambda you wrote down here and explicitly give it for unsupported functions. Then you wouldn't need any of this machinery where you extract the `nonffi` key and do something special if it isn't found.